### ISSUE TITLE
Track semantic scoping arena allocations in allocation snapshots

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#22781]
## Context

`tasks/track_memory_allocations/allocs_semantic.snap` currently reports `0` for all semantic arena allocation columns.

That is misleading because semantic scoping does allocate into an arena, but not into the parser/task allocator passed to `record_stats_in`. `SemanticBuilder::new()` creates `Scoping::default()`, and `Scoping::default()` creates its own `ScopingCell` with `Allocator::default()`. As a result, `tasks/track_memory_allocations/src/lib.rs` only observes system allocations plus arena activity on the outer parser allocator, while scoping's internal arena allocations are invisible.

Current symptom:

```text
File       ... || Arena allocs | Arena reallocs
checker.ts ... ||            0 |              0
...
```

## Goal

Make the semantic allocation snapshot track how many arena allocations and reallocations scoping performs during semantic analysis.

## Possible implementation

Expose allocation stats from `ScopingCell` / `Scoping` behind the existing allocation-tracking cfg used by `oxc_allocator`:

```rust
#[cfg(all(feature = "track_allocations", not(feature = "disable_track_allocations")))]
pub fn allocator_allocation_stats(&self) -> (usize, usize) {
    self.0.borrow_owner().get_allocation_stats()
}
```

Then update `tasks/track_memory_allocations` so the semantic row records the allocation stats from the `Semantic` / `Scoping` returned by `SemanticBuilder`, rather than only reading the outer parser allocator.

Keep the API hidden or cfg-gated if it should only exist for `tasks/track_memory_allocations`.

## Acceptance criteria

- `allocs_semantic.snap` shows non-zero arena allocation counts for semantic fixtures where scoping creates bindings/references.
- The tracked numbers come from scoping's internal allocator, not the parser allocator.
- The existing allocation tracking task updates the semantic snapshot correctly.
- `--all-features` behavior remains compatible with the existing `disable_track_allocations` / `is_all_features` setup.

**Comment by camc314** [comment]
We briefly discussed this in the mtg today. Some concerns were around different numbers of hashmap allocations on different platforms. This can cause issues as the snapshot will be different locally vs in CI.

A potential solution to this is to compile hashbrown without some certain features to ensure that it is aligned completely in different architectures etc
